### PR TITLE
feat: Optionally remove single-child top-level directory from extracted archives

### PR DESCRIPTION
**Pull request by josephine-funken**
## Description

Removes top-level directories from extracted files if its only child is a directory.
This top-level directory in the archive is unnecessary and creates longer file paths than needed.

Fixes issue #401 

## Implemented changes

- [ ] Remove top-level directories by default in `Dataset.extract()`, `dataset_download.extract_dataset()` and `utils.archives.extract_archive()`.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [ ] `tests/utils/archives_test.test_extract_archive_destination_path_None()`
- [ ] `test_extract_archive_destination_path_not_None()`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by SiQube** [comment]
@josephine-funken thank you for this great work! there seems to be a merge conflict which needs to be removed first -- if you have any questions how to resolve it feel free to reach out.

**Comment by josephine-funken** [comment]
Added functionality to remove top-level directories to `downloads.download_and_extract_archive()` and added parameter to corresponding test.

File: src/pymovements/utils/archives.py [hunk 1]
```diff
@@ -88,6 +91,22 @@ def extract_archive(
...
+    if remove_top_level:
+        children = [str(file) for file in destination_path.glob('*')]
+        # Check if top-level directory has a single child
+        if len(children) == 1:
+            single_child = children[0]
+            if Path.is_dir(Path(single_child)):
+                shutil.copytree(single_child, destination_path, dirs_exist_ok=True)
+                shutil.rmtree(single_child)
+        # Check if top-level directory has just the two children archive and extracted archive
+        elif len(children) == 2 and destination_path == source_path.parent:
```
Comment (dkrako) [inline src/pymovements/utils/archives.py:103]: this branch doesn't seem to be covered by tests.
maybe it makes sense to combine the `elif` condition into the `if` a few lines before?

**Review by dkrako — CHANGES_REQUESTED** [review]
Thank you very much for this work!

However I would like to request two changes.

- [x] change the file-handling from copy/remove to move/remove
- [x] the `elif` clause can probably be refactored into the if clause which would hopefully raise coverage to 100%

File: src/pymovements/utils/archives.py [hunk 2]
```diff
@@ -88,6 +91,22 @@ def extract_archive(
     if remove_finished:
         source_path.unlink()
 
+    if remove_top_level:
+        children = [str(file) for file in destination_path.glob('*')]
+        # Check if top-level directory has a single child
+        if len(children) == 1:
+            single_child = children[0]
+            if Path.is_dir(Path(single_child)):
+                shutil.copytree(single_child, destination_path, dirs_exist_ok=True)
```
[the remark this answers is not in the document]
Comment (josephine-funken) [reply src/pymovements/utils/archives.py]: I applied this change

**Review by dkrako — APPROVED** [review]
Great clean work! Thanks a lot!

**Pull request merged**